src/main/groovy/org/shipkit/internal/gradle/ReleaseNotesPlugin.java

```diff
@@ -55,24 +52,23 @@ public void execute(final ReleaseNotesFetcherTask t) {
 
         final AllContributorsFetcherTask contributorsFetcher = (AllContributorsFetcherTask) project.getTasks().getByName(ContributorsPlugin.FETCH_ALL_CONTRIBUTORS_TASK);
 
-        TaskMaker.task(project, UPDATE_NOTES_TASK, IncrementalReleaseNotes.UpdateTask.class, new Action<IncrementalReleaseNotes.UpdateTask>() {
-            public void execute(final IncrementalReleaseNotes.UpdateTask t) {
-                t.setDescription("Updates release notes file.");
-                configureDetailedNotes(t, releaseNotesFetcher, project, conf, contributorsFetcher);
-                GitPlugin.registerChangesForCommitIfApplied(
-                        Arrays.asList(project.file(conf.getReleaseNotes().getFile())), "release notes updated", t);
-            }
-        });
+        TaskMaker.task(project, UPDATE_NOTES_TASK, UpdateReleaseNotesTask.class, new Action<UpdateReleaseNotesTask>() {
+            public void execute(final UpdateReleaseNotesTask t) {
+                t.setDescription("Updates release notes file. Run with '-Ppreview' if you only want to see the preview.");
 
-        TaskMaker.task(project, PREVIEW_NOTES_TASK, IncrementalReleaseNotes.PreviewTask.class, new Action<IncrementalReleaseNotes.PreviewTask>() {
-            public void execute(final IncrementalReleaseNotes.PreviewTask t) {
-                t.setDescription("Shows new incremental content of release notes. Useful for previewing the release notes.");
                 configureDetailedNotes(t, releaseNotesFetcher, project, conf, contributorsFetcher);
+
+                if(!t.isInPreviewMode()){
+                    File releaseNotesFile = project.file(conf.getReleaseNotes().getFile());
+                    GitPlugin.registerChangesForCommitIfApplied(
+                        Arrays.asList(releaseNotesFile), "release notes updated", t);
+                    t.getOutputs().file(releaseNotesFile);
```

Comment:
This is super smart!!! I forgot that we can configure the outputs programatically.

I suggest to make the conditional logic simpler. In the plugin we could do something like:

```
if (project.hasProperty("preview")) {
   t.setPreview(true);
   t.getOutputs().file(t.getReleaseNotesFile());
}
```